rollup-sequencer/src/utils/index.ts

```diff
@@ -187,13 +187,26 @@ async function getSelectedSequencerWithRights(
 	const selectedSequencerMap =
     await apiAt.query.sequencerStaking.selectedSequencer();
   const selectedMap = JSON.parse(selectedSequencerMap.toString());
-  const selectedSequencer = selectedMap[L1_CHAIN].toLowerCase();
-
-  const isSequencerSelected = selectedSequencer === collatorAddress.toLowerCase();
-  const sequencerRights = await apiAt.query.rolldown.sequencersRights(L1_CHAIN);
-  let rights = JSON.parse(sequencerRights.toString())[collatorAddress.toLowerCase()]
-  const hasReadRights = rights.readRights > 0;
-  const hasCancelRights = rights.cancelRights > 0;
+  const selectedSequencerRaw = selectedMap[L1_CHAIN];
+
+  var selectedSequencer;
+  var isSequencerSelected;
+  var hasReadRights;
+  var hasCancelRights;
+
+  if (selectedSequencerRaw === undefined) {
```

Comment:
In general its good to not use if/else.

You can initalize the variables first and 

  var selectedSequencer;
  var isSequencerSelected;
  var hasReadRights;
  var hasCancelRights;
  
  and use 
  
  if selectedSequencerRaw !== undefined
  
  instead to simplify the code and have only IF block and no ELSE block